detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -100,6 +107,43 @@ export class ContainerDetector implements Detector {
     return containerIdStr || '';
   }
 
+  private _isInKubernetesEnvironment(): boolean {
+    return process.env.KUBERNETES_SERVICE_HOST !== undefined;
+  }
+
+  private async _getContainerIdK8(): Promise<string> {
+    const kubeconfig = new KubeConfig();
+    kubeconfig.loadFromDefault();
+
+    const api = kubeconfig.makeApiClient(CoreV1Api);
+    const namespace: string = process.env.NAMESPACE || 'default';
+    const containerName: string | undefined = process.env.CONTAINER_NAME;
+    if (!containerName) {
+      throw new Error('Container name not specified in environment');
+    }
+
+    const response = await api.listNamespacePod(namespace);
```

Comment:
This is a typo, should be `api.listNamespacedPod`. I'm not sure why typescript isn't complaining